fastapi_auth_jwt/dependencies.py

```diff
@@ -0,0 +1,247 @@
+            str | None,
+            Depends(auth_jwt_default_validator_name),
+        ],
+        env: Annotated[
+            Environment,
+            Depends(odoo_env),
+        ],
+    ) -> Payload | None:
+        validator = _get_auth_jwt_validator(
+            self.validator_name or default_validator_name, env
+        )
+        if self.allow_unauthenticated and not _request_has_authentication(
+            request, authorization_header, validator
+        ):
+            return None
+        return _get_jwt_payload_and_validator(
+            request, response, authorization_header, validator
+        )[0]
+
+
+class AuthJwtPartner(BaseAuthJwt):
+    def __call__(
+        self,
+        request: Request,
+        response: Response,
+        authorization_header: Annotated[
+            str | None,
+            Depends(auth_jwt_http_header_authorization),
+        ],
+        default_validator_name: Annotated[
+            str | None,
+            Depends(auth_jwt_default_validator_name),
+        ],
+        env: Annotated[
+            Environment,
+            Depends(odoo_env),
+        ],
+    ) -> Partner:
+        validator = _get_auth_jwt_validator(
+            self.validator_name or default_validator_name, env
+        )
+        if self.allow_unauthenticated and not _request_has_authentication(
+            request, authorization_header, validator
+        ):
+            return env["res.partner"].with_user(env.ref("base.public_user")).browse()
+        payload, validator = _get_jwt_payload_and_validator(
+            request, response, authorization_header, validator
+        )
+        try:
+            uid = validator._get_and_check_uid(payload)
+            partner_id = validator._get_and_check_partner_id(payload)
+        except Unauthorized as e:
+            raise HTTPException(status_code=HTTP_401_UNAUTHORIZED) from e
+        if not partner_id:
+            _logger.info("Could not determine partner from JWT payload.")
+            raise HTTPException(status_code=HTTP_401_UNAUTHORIZED)
+        return env["res.partner"].with_user(uid).browse(partner_id)
+
+
+class AuthJwtOdooEnv(BaseAuthJwt):
+    def __call__(
+        self,
+        request: Request,
+        response: Response,
+        authorization_header: Annotated[
+            str | None,
+            Depends(auth_jwt_http_header_authorization),
+        ],
+        default_validator_name: Annotated[
+            str | None,
+            Depends(auth_jwt_default_validator_name),
+        ],
+        env: Annotated[
+            Environment,
+            Depends(odoo_env),
+        ],
+    ) -> Environment:
+        validator = _get_auth_jwt_validator(
+            self.validator_name or default_validator_name, env
+        )
+        payload, validator = _get_jwt_payload_and_validator(
+            request, response, authorization_header, validator
+        )
+        uid = validator._get_and_check_uid(payload)
+        return odoo_env(user=uid)
```

Comment:
```suggestion
        return env(user=uid)
```
@dnplkndll small error here. Thanks for the migration!